client/test/fund.go

```diff
@@ -0,0 +1,160 @@
+// Copyright 2022 - See NOTICE file for copyright holders.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package test
+
+import (
+	"context"
+	"errors"
+	"math/big"
+	"math/rand"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"perun.network/go-perun/channel"
+	"perun.network/go-perun/client"
+	"perun.network/go-perun/wire"
+	"polycry.pt/poly-go/test"
+)
+
+// FundSetup represents the fund recovery test parameters.
+type FundSetup struct {
+	ChallengeDuration uint64
+	FridaInitBal      channel.Bal
+	FredInitBal       channel.Bal
+	BalanceDelta      channel.Bal
+}
+
+// TestFundRecovery performs a test of the fund recovery functionality for the
+// given setup.
+func TestFundRecovery(ctx context.Context, t *testing.T, params FundSetup, setup func(*rand.Rand) ([2]RoleSetup, channel.Asset)) { //nolint:revive // test.Test... stutters but this is OK in this special case.
```

Comment:
Since the method signature is quite long I would probably wrap it at some point but that's up to you.

Reply:
Done